include/quic/btstream.hpp

```diff
@@ -29,23 +29,22 @@ namespace oxen::quic
 
       private:
         int64_t req_id;
-        bstring data;  // Wrapped in a shared pointer because: a) we store views into this and need
-                       // them to stay valid across moves; and b) we want message to be copyable so
-                       // that it can be captured in lambda args.
-        std::string_view req_type;
-        std::string_view ep;
-        std::string_view req_body;
+        bstring data;
+
+        // We keep the locations of variables fields as relative positions inside `data` *rather*
+        // than using std::string_view members because the string_views are more difficult to
+        // maintain when the object gets moved or copied.
+        using substr_location = std::pair<std::ptrdiff_t, std::size_t>;
```

Comment:
This would be a really cool idea for a PR to oxen-encoding, basically extracting a map or some referential structure indicating the indexes of each data field in the bt-dict or list